### PR TITLE
Avoid some allocations

### PR DESCRIPTION
**Pull request by ghost**
Hello, I've taken a closer look at the Windows implementation of `is_executable` and I think this can be optimized. ~~First, I believe the `to_string_lossy` can simply be removed from both `extension` and `pathext` and shouldn't affect the comparison but I could be wrong on that one so please test this if you are able to.~~
As for the `.map(|ext| ext[1..].to_string())`, I was able to remove the `to_string` by simply merging the cut-off into the `any`. The `collect` is unnecessary and the `any` iteration can simply be done immediately instead of collecting into a `Vec` first.

### CODE REVIEW

**Comment by ghost** [comment]
I'm not entirely sure why it fails but I'm going to close this for now.

File: src/lib.rs [hunk 1]
```diff
@@ -113,21 +113,20 @@ mod windows {
             // Check using file extension
             if let Some(pathext) = std::env::var_os("PATHEXT") {
                 if let Some(extension) = self.extension() {
-                    // Restructure pathext as Vec<String>
+                    let extension = extension.to_string_lossy();
+                    
+                    // Originally taken from:
                     // https://github.com/nushell/nushell/blob/93e8f6c05e1e1187d5b674d6b633deb839c84899/crates/nu-cli/src/completion/command.rs#L64-L74
                     let pathext = pathext
```
Comment (alpha-tango-kilo) [inline src/lib.rs]: This is the only part that's not quite right. You removed the original return statement I had and haven't provided it again, meaning the program always falls back to the old logic (lines 132 onwards of your branch)

```diff
diff --git a/src/lib.rs b/src/lib.rs
index 8b1c664..dfc71e5 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -117,7 +117,7 @@ mod windows {
                     
                     // Originally taken from:
                     // https://github.com/nushell/nushell/blob/93e8f6c05e1e1187d5b674d6b633deb839c84899/crates/nu-cli/src/completion/command.rs#L64-L74
-                    let pathext = pathext
+                    return pathext
                         .to_string_lossy()
                         .split(';')
                         // Filter out empty tokens and ';' at the end
```

This'll make tests pass and reduce allocations as you intended

**Review by alpha-tango-kilo — COMMENTED** [review]
Hiya, not a maintainer or the owner but I noticed you submitted this PR optimising code I contributed. It's a really quick fix for you to apply then hopefully we can get this merged!

When I originally wrote the extension-checking part of the program I was still pretty novice-level at Rust, so I appreciate you coming back and improving on my work

File: src/lib.rs (hunk 1 above)
Comment (ghost) [reply src/lib.rs]: Oh, hello! Alright, let's try that.

**Comment by alpha-tango-kilo** [comment]
Let's get this merged! @fitzgen 

If you wanted help maintaining the project, I'd be happy to offer my services :)

**Review by fitzgen — APPROVED** [review]

**Pull request merged**